code/modules/hydroponics/gene_modder.dm

```diff
@@ -379,38 +255,103 @@
 						QDEL_NULL(seed)
 						update_icon(UPDATE_OVERLAYS)
 				if("replace")
-					if(disk && disk.gene && istype(disk.gene, G.type) && istype(G, /datum/plant_gene/core))
-						seed.genes -= G
+					if(disk && disk.gene && istype(disk.gene, target.type) && istype(target, /datum/plant_gene/core))
+						seed.genes -= target
 						var/datum/plant_gene/core/C = disk.gene.Copy()
 						seed.genes += C
 						C.apply_stat(seed)
 						repaint_seed()
-				if("insert")
-					if(disk && disk.gene && !istype(disk.gene, /datum/plant_gene/core) && disk.gene.can_add(seed))
-						seed.genes += disk.gene.Copy()
-						if(istype(disk.gene, /datum/plant_gene/reagent))
-							seed.reagents_from_genes()
-						disk.gene.apply_vars(seed)
-						repaint_seed()
-
 			update_genes()
-			operation = ""
+			SStgui.update_uis(src)
 			target = null
-	else if(href_list["abort"])
-		operation = ""
-		target = null
-	else if(href_list["set_v"])
-		if(!seed)
 			return
-		seed.variant_prompt(usr, src)
-	else if(href_list["del_v"])
-		if(!seed)
-			return
-		seed.variant = null
-		seed.apply_variant_name()
-		to_chat(usr, "<span class='notice'>You remove the [seed.plantname]'s variant designation.</span>")
 
-	interact(usr)
+	target = seed?.get_gene(params["id"])
+
+	switch(action)
+		if("eject_seed")
+			if(seed)
+				seed.forceMove(loc)
+				user.put_in_hands(seed)
+				seed = null
+				update_genes()
+				update_icon(UPDATE_OVERLAYS)
+			else
+				var/obj/item/I = user.get_active_hand()
+				if(istype(I, /obj/item/seeds))
+					add_seed(I, user)
+			SStgui.update_uis(src)
```

Comment:
try your UI without `SStgui.update_uis(src)` anywhere. TGUI by default should be set to auto update/refresh the UI so I don't believe these calls are necessary (double check tho).